src/main/scala/fi/oph/koski/kela/KelaServlet.scala

```diff
@@ -30,6 +32,22 @@ class KelaServlet(implicit val application: KoskiApplication) extends ApiServlet
       }
     }()
   }
+
+  get("/versiohistoria/:opiskeluoikeusOid") {
+    renderOption[List[OpiskeluoikeusHistoryPatch]](KoskiErrorCategory.notFound.opiskeluoikeuttaEiLöydyTaiEiOikeuksia) {
+      kelaService.opiskeluoikeudenHistoria(getStringParam("opiskeluoikeusOid"))
+    }
+  }
+
+  get("/versiohistoria/:oppijaOid/:opiskeluoikeusOid/:version") {
+    val oppijaOid = HenkilöOid.validateHenkilöOid(getStringParam("oppijaOid"))
+    val opiskeluoikeusOid = OpiskeluoikeusOid.validateOpiskeluoikeusOid(getStringParam("opiskeluoikeusOid"))
+    val oppija = oppijaOid.flatMap(oppija => opiskeluoikeusOid.map(opiskeluoikeus => (oppija, opiskeluoikeus)))
```

Comment:
Olen varmaan väärässä (en ole kauhean paljon Scalaa koodannut), mutta jotenkin tuntuu, että tällainen Either:eiden flatMappäily ois selkeempi koodin lukijalle for-syntaksilla, käsittääkseni tämä koko funktio hoituisi näin:
```
    val oppija = for {
      oppijaOid <- HenkilöOid.validateHenkilöOid(getStringParam("oppijaOid"))
      opiskeluoikeusOid <- OpiskeluoikeusOid.validateOpiskeluoikeusOid(getStringParam("opiskeluoikeusOid"))
      oppija <- kelaService.findKelaOppijaVersion(oppijaOid, opiskeluoikeusOid, getIntegerParam("version"))
    } yield oppija

    renderEither(oppija)
```